ui/storybook/README.md

```diff
@@ -118,13 +118,50 @@ By default, visual tests will be taken for every story in storybook. You can dis
 
 By default, visual tests will be taking for both light mode and dark mode. You can modify this by setting `parameters.happo.themes` to an array of the themes you want (e.g. `['light']` to just use light mode). This is a customization we configured on top of happo's built in behavior, so this is not mentioned in their documentation.
 
+#### Waiting for stable canvas
+
+Happo does its best to wait for the page to stabilize before taking a screenshot, but it does not do anything special for canvas. We cannot use happo's built-in `waitFor` to work around this because it is synchronous and constantly looping to check the status will deny the underlying code the ability to actually modify the canvas.
+
+To work around this, we've added an async `waitForAsync` helper that you can use in concert with an internal `waitForStableCanvas` utility in the storybook package to attempt to wait for a stable canvas element before taking a screenshot.
+
+```ts
+import { waitForStableCanvas } from '@perses-dev/storybook';
+
+const meta: Meta<typeof LineChart> = {
+  component: LineChart,
+  parameters: {
+    happo: {
+      waitForAsync: () => {
+        return waitForStableCanvas('canvas');
+      },
+    },
+  },
+};
+```
+
 ### Happo issues
 
 We have some customization over the default `happo-storybook` configuration related to some differences in our codebase.
 
 - We explicitly build storybook before running happo and have `usePrebuiltPackage` set to `true`. This is needed because the name of the executable for storybook changed in v7 and happo hasn't updated to account for it yet.
 - We need to set explicitly set the `HAPPO_COMMAND` env var when calling `happo-ci-github-actions` to run in CI. Their default script looks in the wrong location because it doesn't fully account for the complexity of a monorepo setup like ours.
-- We use a modified version of the [`register.js`](https://github.com/happo/happo-plugin-storybook/blob/master/src/register.js) file provided by `happo-plugin-storybook`. This enables us to support Storybook v7 and to automatically grab light and dark mode screenshots without creating special stories for them.
+- We use a modified version of [`register.js`](https://github.com/happo/happo-plugin-storybook/blob/master/src/register.js) file provided by `happo-plugin-storybook`. This enables us to add custom `waitForAsync` functionality.
```

Comment:
Custom register.js is baaaaack. I will again follow-up with the happo folks to see if they can bake full support for this in their codebase. They've been really responsive, so hopefully it'll be able to go back away again in the future. I've been really happy with this workflow where I can figure out a fix locally without fully waiting for a change on their end.